### ISSUE TITLE
Java exception when unlocking database

### ISSUE THREAD

**Issue by ascended121** [nasa/CCDD#11]
When opening CCDD, receive message: 
`Cannot connect to project database '######' as user '######'; cause 'database is locked'`

In response, I go to the menu and select Project > Unlock and click the 'Unlock' button. Gui doesn't respond and terminal shows the error message below. Restarting CCDD/repeating the process yields the same message.

Below is an image of the unlock dialog. According to the user guide, the dialog is supposed to list projects which may be unlocked, but mine is not.
![image](https://user-images.githubusercontent.com/18016749/41064550-dcfb3018-69a9-11e8-973c-0b5c73ab880f.png)

This may explain why the dialog validation is failing.

System Info:
OS: Win10
Java: Java 8 Update 171 (64 bit)
PostgreSQL: PostgreSQL 10

```
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException
        at CCDD.CcddDialogHandler.getCheckBoxSelected(CcddDialogHandler.java:322)
        at CCDD.CcddDbManagerDialog.verifySelection(CcddDbManagerDialog.java:776)
        at CCDD.CcddDialogHandler$1.closeWindow(CcddDialogHandler.java:220)
        at CCDD.CcddButtonPanelHandler$1.actionPerformed(CcddButtonPanelHandler.java:305)
        at javax.swing.AbstractButton.fireActionPerformed(Unknown Source)
        at javax.swing.AbstractButton$Handler.actionPerformed(Unknown Source)
        at javax.swing.DefaultButtonModel.fireActionPerformed(Unknown Source)
        at javax.swing.DefaultButtonModel.setPressed(Unknown Source)
        at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(Unknown Source)
        at java.awt.Component.processMouseEvent(Unknown Source)
        at javax.swing.JComponent.processMouseEvent(Unknown Source)
        at java.awt.Component.processEvent(Unknown Source)
        at java.awt.Container.processEvent(Unknown Source)
        at java.awt.Component.dispatchEventImpl(Unknown Source)
        at java.awt.Container.dispatchEventImpl(Unknown Source)
        at java.awt.Component.dispatchEvent(Unknown Source)
        at java.awt.LightweightDispatcher.retargetMouseEvent(Unknown Source)
        at java.awt.LightweightDispatcher.processMouseEvent(Unknown Source)
        at java.awt.LightweightDispatcher.dispatchEvent(Unknown Source)
        at java.awt.Container.dispatchEventImpl(Unknown Source)
        at java.awt.Window.dispatchEventImpl(Unknown Source)
        at java.awt.Component.dispatchEvent(Unknown Source)
        at java.awt.EventQueue.dispatchEventImpl(Unknown Source)
        at java.awt.EventQueue.access$500(Unknown Source)
        at java.awt.EventQueue$3.run(Unknown Source)
        at java.awt.EventQueue$3.run(Unknown Source)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
        at java.awt.EventQueue$4.run(Unknown Source)
        at java.awt.EventQueue$4.run(Unknown Source)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
        at java.awt.EventQueue.dispatchEvent(Unknown Source)
        at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
        at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
        at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
        at java.awt.WaitDispatchSupport$2.run(Unknown Source)
        at java.awt.WaitDispatchSupport$4.run(Unknown Source)
        at java.awt.WaitDispatchSupport$4.run(Unknown Source)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.awt.WaitDispatchSupport.enter(Unknown Source)
        at java.awt.Dialog.show(Unknown Source)
        at java.awt.Component.show(Unknown Source)
        at java.awt.Component.setVisible(Unknown Source)
        at java.awt.Window.setVisible(Unknown Source)
        at java.awt.Dialog.setVisible(Unknown Source)
        at CCDD.CcddDialogHandler.createDialog(CcddDialogHandler.java:2221)
        at CCDD.CcddDialogHandler.showOptionsDialog(CcddDialogHandler.java:778)
        at CCDD.CcddDbManagerDialog$1.complete(CcddDbManagerDialog.java:474)
        at CCDD.CcddBackgroundCommand$1.done(CcddBackgroundCommand.java:156)
        at javax.swing.SwingWorker$5.run(Unknown Source)
        at javax.swing.SwingWorker$DoSubmitAccumulativeRunnable.run(Unknown Source)
        at sun.swing.AccumulativeRunnable.run(Unknown Source)
        at javax.swing.SwingWorker$DoSubmitAccumulativeRunnable.actionPerformed(Unknown Source)
        at javax.swing.Timer.fireActionPerformed(Unknown Source)
        at javax.swing.Timer$DoPostEvent.run(Unknown Source)
        at java.awt.event.InvocationEvent.dispatch(Unknown Source)
        at java.awt.EventQueue.dispatchEventImpl(Unknown Source)
        at java.awt.EventQueue.access$500(Unknown Source)
        at java.awt.EventQueue$3.run(Unknown Source)
        at java.awt.EventQueue$3.run(Unknown Source)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
        at java.awt.EventQueue.dispatchEvent(Unknown Source)
        at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
        at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
        at java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
        at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
        at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
        at java.awt.EventDispatchThread.run(Unknown Source)
```

**Comment by KevinMcCluney** [comment]
Please provide the CCDD version (from the main window header or the Help | About dialog).  The lock status implies the database is in use by another instance of CCDD (perhaps one that terminated abnormally - that's the case for which the unlock command was designed to be used).  However, use of the Unlock button shouldn't cause an exception.  Were you able to access the database in question prior to this?  Are other databases accessible?  I haven't tested the application on the OS, Java, or PostgreSQL versions you cited - I'll attempt to set that up, but it'll be later this week.

**Comment by ascended121** [comment]
From the about dialog:
CCDD: 1.2.33 6-5-2018
Java: 1.8.0_171 (64bit)
JDBC: PostgreSQL 9.4.1207.jre7 (type 4)
Jetty: 9.2.18.v20160721

Yes, was able to add tables, create groups, associate scripts and execute them prior to this happening. I was playing with the execute argument and did have to terminate a script, so its likely that's how the database became locked. Yes, I'm able to create a new test project and connect to it, so it looks like its just an issue with my main project being locked.

Thanks, appreciate the attention.

**Comment by KevinMcCluney** [comment]
Thanks for the extra explanation and the Unlock dialog snapshot.  This indicates that an exception condition is occurring when the Unlock dialog is created, but the stack trace isn't being output.  I uploaded a new version of CCDD.jar that captures this sort of exception when opening the Unlock dialog.  When you get a chance, download it and repeat the steps to unlock the database.  I expect that a dialog will appear for the unexpected exception - paste an image of the error dialog in a reply.

**Comment by ascended121** [comment]
This dialog popped up as soon as I attempted to open the unlock dialog:

![image](https://user-images.githubusercontent.com/18016749/41119036-b890a468-6a5f-11e8-9231-d10dcb5b07bf.png)

No exception appeared at the command line this time. After clicking ok the dialog did not appear. Let me know if I can collect any additional info to help!

**Comment by KevinMcCluney** [comment]
The version uploaded to Github now has some code that should get past the issue.  I can't yet determine the root cause, but your information has helped to narrow down where the failure occurs.  When the Unlock dialog is displayed you'll get one of two results: (1) a dialog will appear stating 'Cannot retrieve ACTIVE_BY_USER list; cause 'list query returned null ResultSet', or (2) the command line will display 'ResultSet string is null'.  Whichever is the case the issue stems from a failure to retrieve information (database and user) for the databases that are currently open - please let me know if you get either (or none) of these.
The code change may allow the unlock to be completed, but there is something else going on.

**Comment by KevinMcCluney** [comment]
Did any dialog or message appear when the Unlock button was pressed?  If it's working correctly now I'll close this issue.

**Comment by ascended121** [comment]
No additional dialogs were generated and the database was unlocked, so it appears to be working as intended. Thanks!

**Comment by ascended121** [comment]
This issue appears to have resurfaced, though with a slightly different stack trace. Observed on 1.4.27. After this exception the unlock menu displays "Locked by *unknown*".

```
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException
        at CCDD.CcddDbControlHandler.getQuotedName(CcddDbControlHandler.java:635)
        at CCDD.CcddDbControlHandler.buildDatabaseCommentCommand(CcddDbControlHandler.java:1230)
        at CCDD.CcddDbControlHandler.setDatabaseLockStatus(CcddDbControlHandler.java:1075)
        at CCDD.CcddDbManagerDialog$1.complete(CcddDbManagerDialog.java:761)
        at CCDD.CcddBackgroundCommand$1.done(CcddBackgroundCommand.java:192)
        at javax.swing.SwingWorker$5.run(Unknown Source)
        at javax.swing.SwingWorker$DoSubmitAccumulativeRunnable.run(Unknown Source)
        at sun.swing.AccumulativeRunnable.run(Unknown Source)
        at javax.swing.SwingWorker$DoSubmitAccumulativeRunnable.actionPerformed(Unknown Source)
        at javax.swing.Timer.fireActionPerformed(Unknown Source)
        at javax.swing.Timer$DoPostEvent.run(Unknown Source)
        at java.awt.event.InvocationEvent.dispatch(Unknown Source)
        at java.awt.EventQueue.dispatchEventImpl(Unknown Source)
        at java.awt.EventQueue.access$500(Unknown Source)
        at java.awt.EventQueue$3.run(Unknown Source)
        at java.awt.EventQueue$3.run(Unknown Source)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
        at java.awt.EventQueue.dispatchEvent(Unknown Source)
        at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
        at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
        at java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
        at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
        at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
        at java.awt.EventDispatchThread.run(Unknown Source)
```

**Comment by KevinMcCluney** [comment]
This is corrected in version 1.4.28.
The issue is an artifact of the recent change that allows using PostgreSQL keywords for database, table, and column names.
